source/components/triggering-changes-with-actions.md

```diff
@@ -0,0 +1,277 @@
+
+Next, how do we detect that a user has double clicked our component?
+Ember has built-in support for common browser events like this. In our
+component's JavaScript, we can implement a method called
+`doubleClick()`. Every time the user double clicks this component, Ember
+will call the `doubleClick()` method.
+
+```app/components/double-click-button.js
+export default Ember.Component.extend({
+  tagName: 'button',
+
+  // This method is called every time the user double clicks the
+  // component.
+  doubleClick() {
+  }
+});
+```
+
+Now that we've implemented the `doubleClick()`  method, we still need to
+trigger the `userDidDeleteAccount` action that we've been given.  Let's
+do that now.
+
+When you provide an action, it get turned into a method on the component
+you can call. Above we said:
+
+```hbs
+onDoubleClick=(action 'userDidDeleteAccount')`
+```
+
+That means that calling the component's `onDoubleClick` method will call
+the parent's `userDidDeleteAccount` action.
+
+```app/components/double-click-button.js
+doubleClick() {
+  // Call the component's `onDoubleClick()` method to trigger
+  // the parent's `userDidDeleteAccount` action.
+  this.onDoubleClick();
+}
+```
+
+Actions are **just functions** you can call, like any other method on your component.
+
+That's it! Under the hood, `onDoubleClick` is a function that knows how
+to access the parent component and trigger its `userDidDeleteAccount`
+action. When the user double clicks your component, the component will
+call `this.onDoubleClick()`, which will trigger the parent component's
+`userDidDeleteAccount` action, deleting the account.
+
+### Thinking About Actions
+
+#### Actions and Attributes
+
+Like normal attributes, actions set a property on the component; the
+only difference is that the property is set to a function that knows how
+to trigger behavior.
+
+That makes it easy to remember how to add an action to a component. It's
+just like passing an attribute, but you use the `action` helper to pass
+a function instead.
+
+In the future versions of our `double-click-button` component, for
+example, we may want to be able to pass a `title` attribute. You can see
+how similar the syntax is:
+
+```app/component/user-profile.hbs
+{{double-click-button
+  title="Delete Account"
+  onDoubleClick=(action 'userDidDeleteAccount')}}
+```
+
+#### Actions and Callbacks
+
+You may have noticed that component actions are similar to another
+concept in JavaScript: **callbacks**. You can think of actions as being
+like callbacks for components.
+
+In JavaScript, you can pass a callback to a function that it can invoke
+at a later time. For example, we might write a helper function that takes
+a callback and adds it as an event listener to an element:
+
+```js
+function onClick(element, onClickFunc) {
+  element.addEventListener('click', onClickFunc);
+}
+
+let element = document.getElementById('delete-button');
+let displayAlert = () => {
+  alert("button clicked!");
+};
```

Comment:
Should we use non-ES6 syntax here when demonstrating older callback syntax?  Callbacks are used heavily by folks in the jQuery world, but ES6 isn't.  Seems like we're trying to help explain actions based on something that is familiar, but using syntax that may be unfamiliar may make that harder to grok ...